cloud-cli beta and SL GA launch (#4191)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/saved-queries.md

---
title: Saved Queries
id: saved-queries
description: "Saved queries are a way to save commonly used queries in MetricFlow. They can be used to save time and avoid writing the same query over and over again."
sidebar_label: "Saved Queries"
tags: [Metrics, Semantic Layer]
---

> Saved queries are a way to save commonly used queries in MetricFlow. You can group metrics, dimensions and filters that are logically related into a saved query. The spec for defining a saved query is below:

Author: ```suggestion
Saved queries are a way to save commonly used queries in MetricFlow. You can group metrics, dimensions and filters that are logically related into a saved query. 

To define a saved query, refer to the following specification:
```

 Parameter | Description | Type |
| --------- | ----------- | ---- |
| `name` | The name of the metric. | Required |
| `description` | The description of the metric. | Optional |
| `metrics` | The metrics included in the saved query. | Required |
| `group_bys` | The value that will be displayed in downstream tools. | Required |
| `where` | Filter applied to the query. | Optional |

> An example of a saved query is below:

Author: ```suggestion
The following is an example of a saved query:
```

```yaml
saved_query:
  name: p0_booking
  description: Booking-related metrics that are of the highest priority.
  metrics:
    - bookings
    - instant_bookings
  group_bys:
    - TimeDimension('metric_time', 'day')
    - Dimension('listing__capacity_latest')
  where:
    - "{{ Dimension('listing__capacity_latest') }} > 3"
```

> FAQs and Usage Notes:

Author: ```suggestion
### FAQs 
```

> * Dimensions refrenced in the group by or where must be common across all metrics in a saved query.

Author: ```suggestion
* All metrics in a saved query need to use the same dimensions in the `group_by` or `where` clauses.
```
